### ISSUE TITLE
Support parsing T-SQL's named DEFAULT constraints in CREATE TABLE

### ISSUE THREAD

**Issue by NisanurBulut** [jOOQ/jOOQ#15142]
### SQL translator incorrectly translates this query:

```sql
create table dbo.OrderLots
(
    OrderLotID  int identity
        constraint PK_OrderLots
            primary key,
    OrderID     int                                  not null
        constraint FK_OrderLots_Orders
            references dbo.Orders,
    LotNumber   int                                  not null,
    PanelCount  int
        constraint DF_OrderLots_PanelCount default 0 not null,
    PanelTypeID int
        constraint FK_OrderLots_PanelTypes
            references dbo.PanelTypes,
    OrderName   nvarchar(100),
    GOIBAT      datetime,
    POIBIT      datetime,
    GOIBIT      datetime,
    PAMT        datetime,
    GAMBT       datetime,
    PFT         datetime,
    PST         datetime,
    CreaDate    smalldatetime
)
go
```

### Parser output:

```sql
CHECK, NOT NULL, NULL, PRIMARY KEY, REFERENCES, or UNIQUE expected: [11:44] ...  int
        constraint DF_OrderLots_PanelCount [*]default 0 not null,
    PanelTypeID int
        constraint FK_OrderLots_PanelT...
```

**Comment by lukaseder** [comment]
Thanks a lot for your report.

It seems that T-SQL supports named `DEFAULT` "constraints", interesting feature. We could support it in jOOQ:

- https://github.com/jOOQ/jOOQ/issues/15146

However, I doubt it can be translated to most dialects. So perhaps, for the time being, we could just parse the constraint name and ignore it, keeping only the `DEFAULT` expression, in `CREATE TABLE`

**Comment by lukaseder** [comment]
Note, there are also implications for `ALTER TABLE` statements (`ADD` and `DROP` constraint), which will be out of scope for this issue here.

As a workaround you can just remove the constraint name declaration.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.19.0. Thanks again for your report.
